### PR TITLE
add custom bind addresss support to init

### PR DESCRIPTION
**Pull request by rockmenjack**
this pull request adds `--bind-address` to specify listen address generated in `etcd.env`

### CODE REVIEW

**Comment by rockmenjack** [comment]
CLA signed

**Comment by dlipovetsky** [comment]
Thank you @rockmenjack!

/ok-to-test

File: apis/config.go [hunk 1]
```diff
@@ -301,10 +304,37 @@ func DefaultAdvertiseClientURLs(cfg *EtcdAdmConfig) error {
 }
 
 // Returns the address associated with the host's default interface.
```
Comment (dlipovetsky) [inline apis/config.go:306]: I think the logic in the function is sound, but this function no longer does what its comment says it does.

```suggestion
// Returns the user-defined address; if that is undefined, returns the address associated with the host's default interface.
```

(I think refactoring this code would be nice, but not required for this PR.)

Comment (rockmenjack) [reply apis/config.go]: pull request updated

**Comment by dlipovetsky** [comment]
@rockmenjack, sorry for the delay, @justinsb and I are looking at the PR at the office hours this morning, and we see that `defaultExternalAddress` function actually does validation of the user-defined address, and returns an error if the validation fails. That code looks like it should belong in a different function. Would you be ok if I added a commit to your PR, or would you like to try doing it yourself?

**Comment by rockmenjack** [comment]
How about just rename the function and update its comment to reflect the changed behavior? (see above commit)

**Pull request merged**